Rakefile

```diff
@@ -23,6 +25,11 @@ task :test do
           next
         end
 
+        if username[0].upcase != filename.upcase
```

Comment:
Maybe `unless username[0].downcase == filename`

Reply:
will `filename` always be lower case?  If that is a certainty then I am fine with this suggestion.  Otherwise wouldn't it be safer to force both sides to be the same case?

Reply:
```suggestion
        unless username.start_with?(filename, filename.upcase)
```